File: frontend-app/profile-page/src/model/UserResponseDTO.ts
import { UserRoleResponseDTO } from './UserRoleResponseDTO';

export class UserResponseDTO {
    id: string;
    name: string;
    surname: string;
    email: string;
    phone: string;
    birthDate: string;
    userRoles: UserRoleResponseDTO[];

    constructor(
        id: string,
        name: string,
        surname: string,
        email: string,
        phone: string,
        birthDate: string,
        userRoles: UserRoleResponseDTO[]
    ) {
        this.id = id;
        this.name = name;
        this.surname = surname;
        this.email = email;
        this.phone = phone;
        this.birthDate = birthDate;
        this.userRoles = userRoles;
    }
}

File: api/user-service/src/main/java/com/carrental/userservice/model/dto/UserResponseDTO.java
package com.carrental.userservice.model.dto;

import lombok.Data;

import java.io.Serializable;
import java.util.List;

@Data
public class UserResponseDTO implements Serializable {

    private Long id;

    private String name;

    private String surname;

    private String email;

    private String phone;

    private String birthDate;

    private List<UserRoleResponseDTO> userRoles;
}
